### PR TITLE
Add some of missed cmath functions

### LINKED ISSUE #16632: Add atan2 function
`atan2` is used when `x` and `y` coordinates are given, in this case it has better performance and wider range than `atan`. See [Wikipedia article](https://en.wikipedia.org/wiki/Atan2) for the full description and motivation

### PR DESCRIPTION
**Pull request by hombit**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=ru

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add new `cmath` functions:
- acosh
- asinh
- atan2
- atanh
- cosh
- hypot
- log1p
- sinh

Fix #16632

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Let's also add a test (`tests/queries/0_stateless`) that proves that these functions exist and give correct result up to 3 digits (`round`).

**Comment by alexey-milovidov** [comment]
2020-11-03 20:29:23 ld.lld-10: error: undefined symbol: DB::registerFunction**AS**inh(DB::FunctionFactory&)
2020-11-03 20:29:23 >>> referenced by registerFunctionsMath.cpp
2020-11-03 20:29:23 >>>               registerFunctionsMath.cpp.o:(DB::registerFunctionsMath(DB::FunctionFactory&)) in archive src/Functions/libclickhouse_functions.a
2020-11-03 20:29:23 >>> did you mean: DB::registerFunction**As**inh(DB::FunctionFactory&)

**Comment by alexey-milovidov** [comment]
https://clickhouse-test-reports.s3.yandex.net/16636/7d363d26954035f8361bf5cffffa019a922d7d5f/style_check/output.txt

**Comment by hombit** [comment]
Sorry for the typo. I fixed it and regenerated `../../src/Functions/ya.make`

**Comment by alexey-milovidov** [comment]
ACos, ATan also?

File: tests/queries/0_stateless/00087_math_functions.sql [hunk 1]
```diff
@@ -82,6 +88,48 @@ select acos(-1) = pi();
 select atan(0) = 0;
 select atan(1) = pi() / 4;
 
+select atan2(0, 1) = 0;
+select atan2(0, 2) = 0;
+select atan2(1, 0) = pi() / 2;
+select atan2(1, 1) = pi() / 4;
+select atan2(-1, -1) = -3 * pi() / 4);
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/00087_math_functions.sql:95]: Unexpected paren.

File: tests/queries/0_stateless/00087_math_functions.sql [hunk 2]
```diff
@@ -56,6 +56,12 @@ select log2(2) = 1;
 select log2(4) = 2;
 select sum(abs(log2(exp2(x)) - x) < 1.0e-9) / count() from system.one array join range(1000) as x;
 
+select log1p(-1) = -inf;
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/00087_math_functions.sql:59]: It's much better to add a new test than to edit existing test.

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@hombit The test does not work.

**Comment by alexey-milovidov** [comment]
Still does not.

**Comment by alexey-milovidov** [comment]
@hombit Need some adjustments in tests.

**Comment by hombit** [comment]
Sorry for the delay, finally I could test it locally and I hope it works now

**Comment by alexey-milovidov** [comment]
Thank you! Merged.

**Pull request merged**